.github/workflows/unit-tests.yml

```
@@ -21,86 +23,76 @@ jobs:
+            phpcs_version: '3.5.7'
             allowed_failure: false
           # Add extra build to test against PHPCS 4.
           - php: '7.4'
-            phpcs_branch: '4.0.x-dev as 3.9.99'
+            phpcs_version: '4.0.x-dev as 3.9.99'
             allowed_failure: true
           - php: '8.1'
-            phpcs_branch: 'dev-master'
+            phpcs_version: 'dev-master'
             allowed_failure: true
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
 
-      - name: Get Composer cache directory
-        id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
-
-      - name: Set up Composer caching
-        uses: actions/cache@v2
-        env:
-          cache-name: cache-composer-dependencies
-        with:
-          path: ${{ steps.composer-cache.outputs.dir }}
-          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-composer-
-
-      - name: Set up PHP
-        uses: shivammathur/setup-php@v2
-        if: ${{ matrix.phpcs_branch == 'dev-master' }}
-        with:
-          php-version: ${{ matrix.php }}
-          coverage: none
-
       # On stable PHPCS versions, allow for PHP deprecation notices.
       # Unit tests don't need to fail on those for stable releases where those issues won't get fixed anymore.
+      - name: Setup ini config
+        id: set_ini
+        run: |
+          if [ "${{ matrix.phpcs_version }}" != "dev-master" ]; then
+            echo '::set-output name=PHP_INI::error_reporting=E_ALL & ~E_DEPRECATED'
+          else
+            echo '::set-output name=PHP_INI::error_reporting=E_ALL'
+          fi
+
       - name: Set up PHP
         uses: shivammathur/setup-php@v2
-        if: ${{ matrix.phpcs_branch != 'dev-master' }}
         with:
           php-version: ${{ matrix.php }}
+          ini-values: ${{ steps.set_ini.outputs.PHP_INI }}
           coverage: none
-          ini-values: error_reporting = E_ALL & ~E_DEPRECATED
+          tools: cs2pr
 
       # Set Composer up to download only PHPCS from source for PHPCS 4.x.
       # The source is needed to get the base testcase from PHPCS.
       # All other jobs can use `auto`, which is Composer's default value.
       - name: Configure PHPCS
-        if: ${{ startsWith( matrix.phpcs_branch, '4' ) }}
+        if: ${{ startsWith( matrix.phpcs_version, '4' ) }}
         run: composer config preferred-install.squizlabs/php_codesniffer source
 
-      - name: Install PHPCS
-        run: composer require squizlabs/php_codesniffer:"${{ matrix.phpcs_branch }}" --no-update --no-scripts
+      - name: Set PHPCS version
+        run: composer require squizlabs/php_codesniffer:"${{ matrix.phpcs_version }}" --no-update --no-scripts
 
       - name: Install Composer dependencies (PHP < 8.0 )
         if: ${{ matrix.php < 8.0 }}
-        run: composer install
+        uses: "ramsey/composer-install@v1"
```

```suggestion
        uses: ramsey/composer-install@v1
```

I think we can omit the quotes.
```suggestion
        uses: "ramsey/composer-install@v1"
```
```suggestion
        uses: "ramsey/composer-install@v1"
```
```suggestion
        uses: "ramsey/composer-install@v1"
```
```suggestion
        uses: "ramsey/composer-install@v1"
```
```suggestion
        uses: "ramsey/composer-install@v1"
```
```suggestion
        uses: "ramsey/composer-install@v1"
```
```suggestion
        uses: "ramsey/composer-install@v1"
```